### PR TITLE
Fix/material UI cleanup

### PR DESCRIPTION
**Pull request by FredrikOseberg**
- General UI/Improvments
- Implemented protected routes and capabilities for standalone routes
- Refactoring
- Began implementing interfaces for users and routes

Later iterations will involve:
- Moving some of this logic serverside, especially ordering of strategies.
- Reset loading counter on logout
- Avoid flashing features page when logging out

### CODE REVIEW

File: src/component/history/EventHistory/EventHistory.jsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { useEffect } from 'react';
```
Comment (chriswk) [inline src/component/history/EventHistory/EventHistory.jsx:1]: Seems like a candidate for tsx? New file and all?

**Review by chriswk — APPROVED** [review]
I see you've moved a couple of components to more semantically correct paths. I like it.

**Comment by ivarconr** [comment]
Not showing "placeholder" for feature toggles when signing in:

https://www.loom.com/share/5780b2497f9241de9518eb874a56e0c4

**Review by ivarconr — COMMENTED** [review]
All in all this looks like a great improvement. I do not have chance to verify all small fixes in here, but my impression is that this is a huge step in closer to v4.

File: src/component/user/Login/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import { useEffect } from 'react';
+import classnames from 'classnames';
+import useMediaQuery from '@material-ui/core/useMediaQuery';
+import { useTheme } from '@material-ui/core/styles';
+import { Typography } from '@material-ui/core';
+
+import AuthenticationContainer from '../authentication-container';
+import ConditionallyRender from '../../common/ConditionallyRender';
+
+import { ReactComponent as SwitchesSVG } from '../../../icons/switches.svg';
```
Comment (ivarconr) [inline src/component/user/Login/Login.jsx:10]: Could we use the Unleash Logo instead, possible an inverted version to make it work with the dark background?

File: src/component/history/EventHistory/EventHistory.jsx (hunk 1 above)
Comment (FredrikOseberg) [reply src/component/history/EventHistory/EventHistory.jsx:1]: Certainly.

File: src/component/user/Login/Login.jsx (hunk 2 above)
Comment (FredrikOseberg) [reply src/component/user/Login/Login.jsx:10]: Yep. I'll add that.

File: src/component/history/EventHistory/EventHistory.jsx (hunk 1 above)
Comment (FredrikOseberg) [reply src/component/history/EventHistory/EventHistory.jsx:1]: Certainly. But I noticed when I started converting App.tsx to type

> Not showing "placeholder" for feature toggles when signing in:
> 
> https://www.loom.com/share/5780b2497f9241de9518eb874a56e0c4

Thanks. Handling that by resetting loading count on logout now.

**Comment by FredrikOseberg** [comment]
> Not showing "placeholder" for feature toggles when signing in:
> 
> https://www.loom.com/share/5780b2497f9241de9518eb874a56e0c4

Thanks. Handling that now by resetting loading count on logout.

**Pull request merged**